crates/transport_iroh/src/lib.rs

```diff
@@ -307,10 +327,16 @@ impl IrohTransport {
         // streams from the remote.
         spawn_connection_reader(ctx.clone(), connections.clone());
 
-        // Perform preflight as first message on the new connection.
+        // Send peer URL to remote first.
```

Comment:
Is there no other way for the remote to get to know the peer url associated to a connection? It seems a bit problematic to have to rely on the peer to tell us their correct peer URL.

Reply:
There are other ways, like a centralized bootstrap server for example. Why does it seem problematic?

Reply:
If we block messages by peer URL but a peer can just send us a frame with a URL update and we'll update their URL on our end then they can easily circumvent blocks by regularly sending updates to their peer URLs I think.

Reply:
And those attempts can be prevented: https://github.com/holochain/kitsune2/pull/402/files#diff-4ef1062a9e6a1da8385e18e81b48be43b2c1bb9861facc27a2af3c0b83847bd7R482-R485